src/components/Slider/index.tsx

```diff
@@ -51,81 +56,97 @@ export const Slider = ({
+    <Container maxW={'8xl'}>
+      <Box
+        sx={mergeWith(
+          // Default pagination styles
+          {
+            '.swiper': {
+              '&-pagination': {
+                top: 0,
+                textAlign: 'right',
+                bottom: 'unset',
+              },
             },
           },
-        },
-        // Override default pagination styles
-        swiperStyles,
-      )}
-    >
-      {heading && (
-        <Heading size="xl" my={6}>
-          {heading}
-        </Heading>
-      )}
-
-      <SwiperBox
-        slidesPerView={responsiveSlidesPerView}
-        spaceBetween={responsiveSpaceBetween}
-        py={6}
-        {...mergeWith({ ...defaultSwiperProps, ...rest })}
+          // Override default pagination styles
+          swiperStyles,
+        )}
       >
-        {/* Render custom children if it's provided instead of default Card list */}
-        {isLoading
-          ? new Array(5).fill(0).map((_, i) => (
-              <SwiperSlide key={i}>
-                <Box borderRadius="lg" boxShadow="sm" overflow="hidden">
-                  <Skeleton h={150} />
-                  <VStack align="stretch" spacing={6} p={4}>
-                    <SkeletonText w={50} noOfLines={1} mt={4} />
-                    <SkeletonText w={150} noOfLines={1} mt={4} />
-                    <SkeletonText noOfLines={3} mt={4} />
-                    <SkeletonText w={100} noOfLines={1} mt={4} />
-                  </VStack>
-                </Box>
-              </SwiperSlide>
-            ))
-          : children?.map((child, i) => (
-              <SwiperSlide key={i}>
-                {({ isActive }) => (
-                  <Box {...(isActive && activeStyles)}>{child}</Box>
-                )}
-              </SwiperSlide>
-            )) ||
-            items?.map((item, i) => (
-              <SwiperSlide key={i}>
-                {({ isActive }) => {
-                  return (
-                    <Card
-                      item={item}
-                      _hover={{
-                        transform: 'translateY(-5px)',
-                        boxShadow: 'lg',
-                      }}
-                      {...customStyles}
-                      {...(isActive && activeStyles)}
-                    />
-                  )
-                }}
-              </SwiperSlide>
-            )) ||
-            'No data found'}
-      </SwiperBox>
-    </Box>
+        {heading && (
+          <Heading size="xl" my={6}>
+            {heading}
+          </Heading>
+        )}
+        {withHero && !isLoading && (
+          <SliderHero items={items} index={activeIndexNumber} />
+        )}
+        <SwiperBox
+          slidesPerView={responsiveSlidesPerView}
+          spaceBetween={responsiveSpaceBetween}
+          py={6}
+          onActiveIndexChange={
+            withHero
+              ? ({ activeIndex }: { activeIndex: number }) =>
+                  onIndexChangeHandler(activeIndex)
```

Comment:
You have already a condition _(depended on `items` which is also outside of the function)_. You may want to move `withHero` next to the `items` condition.

```tsx
onActiveIndexChange={onIndexChangeHandler}
```
```ts
const onIndexChangeHandler = ({ index }) => {
    withHero && items && setActiveIndex((index + 3) % items.length)
  }
```

or
```tsx
onActiveIndexChange={({ activeIndex }) => withHero && items &&
  onIndexChangeHandler(activeIndex) ?? undefined
}
```